media/js/base/mozilla-notification-banner.js

```diff
@@ -0,0 +1,269 @@
+            'event': 'in-page-interaction'
+        });
+    };
+
+    /**
+     * Handles interaction with notification close button and removes from the DOM.
+     */
+    NotificationBanner.close = function() {
+        var notification = document.querySelector('.notification-banner');
+        var close = notification.querySelector('.notification-banner-close');
+
+        if (notification) {
+            close.removeEventListener('click', NotificationBanner.close, false);
+            document.body.removeChild(notification);
+            NotificationBanner.setCookie(NotificationBanner.COOKIE_INTERACTION_VALUE);
+            NotificationBanner.trackGAClose();
+        }
+    };
+
+    /**
+     * Tracks interaction with notification CTA button in GA.
+     */
+    NotificationBanner.trackGAConfirm = function() {
+        window.dataLayer.push({
+            'data-banner-name': _options.name,
+            'data-banner-click': '1',
+            'eAction' : _options.confirmAction,
+            'eLabel': _options.confirmLabel,
+            'event': 'in-page-interaction'
+        });
+    };
+
+    /**
+     * Handles interaction with notification CTA button.
+     */
+    NotificationBanner.confirm = function(e) {
+        // for control + click just set the cookie.
+        if (e.metaKey || e.ctrlKey) {
+            NotificationBanner.setCookie(NotificationBanner.COOKIE_INTERACTION_VALUE);
+            NotificationBanner.trackGAConfirm();
+        }
+        // else redirect after setting the cookie.
+        else {
+            e.preventDefault();
+            NotificationBanner.setCookie(NotificationBanner.COOKIE_INTERACTION_VALUE);
+            NotificationBanner.trackGAConfirm();
+            NotificationBanner.doRedirect(e.target);
+        }
+    };
+
+    /**
+     * Binds click events for notification CTA and close buttons.
+     */
+    NotificationBanner.bind = function() {
+        var confirm = document.querySelector('.notification-banner .notification-banner-confirm');
+        var close = document.querySelector('.notification-banner .notification-banner-close');
+
+        confirm.addEventListener('click', NotificationBanner.confirm, false);
+        close.addEventListener('click', NotificationBanner.close, false);
+    };
+
+    /**
+     * Constructs a notification ready to be inserted into the DOM.
+     * @param {Object} options - Object literal with the same format defined for init()
+     * @return {HTMLNode}
+     */
+    NotificationBanner.create = function(options) {
+        var _notification = document.createDocumentFragment();
+        var _container = document.createElement('div');
+        var _content = document.createElement('div');
+        var _contentContainer = document.createElement('div');
+        var _heading = document.createElement('h2');
+        var _message = document.createElement('p');
+        var _confirm = document.createElement('a');
+        var _close = document.createElement('button');
```

Comment:
sigh yeah... but even if they did this would still be no use for old Firefoxen :/